SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -214,20 +217,48 @@ Interest InterestFactory::CreatePlayerOwnedActorInterest() const
 	return NewInterest;
 }
 
-void InterestFactory::AddUserDefinedQueries(const QueryConstraint& LevelConstraints, TArray<SpatialGDK::Query>& OutQueries) const
+void InterestFactory::AddActorUserDefinedQueries(const AActor* InActor, const QueryConstraint& LevelConstraints, TArray<SpatialGDK::Query>& OutQueries, bool ResurseChildren/*=false*/) const
 {
-	check(Actor);
 	check(ClassInfoManager);
 
+	if (!InActor)
+	{
+		return;
+	}
+
 	TArray<UActorInterestComponent*> ActorInterestComponents;
-	Actor->GetComponents<UActorInterestComponent>(ActorInterestComponents);
+	InActor->GetComponents<UActorInterestComponent>(ActorInterestComponents);
 	if (ActorInterestComponents.Num() == 1)
 	{
 		ActorInterestComponents[0]->CreateQueries(*ClassInfoManager, LevelConstraints, OutQueries);
 	}
 	else if (ActorInterestComponents.Num() > 1)
 	{
-		UE_LOG(LogInterestFactory, Error, TEXT("%s has more than one ActorInterestQueryComponent"), *Actor->GetPathName());
+		UE_LOG(LogInterestFactory, Error, TEXT("%s has more than one ActorInterestComponent"), *InActor->GetPathName());
+		checkNoEntry()
+	}
+
+	if (ResurseChildren)
+	{
+		for (const auto& Child : InActor->Children)
+		{
+			AddActorUserDefinedQueries(Child, LevelConstraints, OutQueries, true);
+		}
+	}
+}
+
+void InterestFactory::AddUserDefinedQueries(const QueryConstraint& LevelConstraints, TArray<SpatialGDK::Query>& OutQueries) const
+{
+	SCOPE_CYCLE_COUNTER(STAT_InterestFactoryAddUserDefinedQueries);
+
+	if (APlayerController* PlayerController = Cast<APlayerController>(Actor))
+	{
+		AddActorUserDefinedQueries(Actor, LevelConstraints, OutQueries, true);
+		AddActorUserDefinedQueries(PlayerController->GetPawn(), LevelConstraints, OutQueries, true);
```

Comment:
Would we be adding the Pawn's queries twice ? Children is defined in terms of Owner, not attachment, AFAIK, so I would expect the first call to recurse into the PlayerController's Pawn, sinc it is supposed to own it.

Reply:
This was my original thought to. When I took a look in example project this did not seem to be the case.